### PR TITLE
🏗 Add catch-all type-check target and src dirs

### PR DESCRIPTION
**Pull request by rcebulko**

### CODE REVIEW

File: build-system/tasks/check-types.js [hunk 1]
```diff
@@ -30,7 +30,12 @@ const {extensions, maybeInitializeExtensions} = require('./extension-helpers');
 const {log} = require('../common/logging');
 const {typecheckNewServer} = require('../server/typescript-compile');
 
-const EXTERNS_GLOB = 'src/core{,/**}/*.extern.js';
+/**
+ * Files that pass type-checking but don't belong to a passing directory target.
```
Comment (samouri) [inline build-system/tasks/check-types.js:34]: Can you add a note that this is purely a temporary measure?

File: build-system/tasks/check-types.js [hunk 2]
```diff
@@ -52,6 +57,76 @@ const getExtensionSrcPaths = () =>
...
+  },
+  'src-inabox': {
+    srcGlobs: ['src/inabox{,/**}/*.js'],
+    warningLevel: 'QUIET',
+  },
+  'src-polyfills': {
+    srcGlobs: ['src/polyfills{,/**}/*.js'],
+    warningLevel: 'QUIET',
+  },
+  'src-polyfillstub': {
```
Comment (samouri) [inline build-system/tasks/check-types.js:93]: nit: I'd combine this with src-polyfills

**Review by samouri — APPROVED** [review]
Thank you for making this!

File: build-system/tasks/check-types.js (hunk 2 above)
Comment (rcebulko) [reply build-system/tasks/check-types.js:93]: Two steps ahead of ya: https://github.com/ampproject/amphtml/pull/34020
Will update whichever one gets submitted second.

File: build-system/tasks/check-types.js [hunk 3]
```diff
@@ -52,6 +57,76 @@ const getExtensionSrcPaths = () =>
  * @type {Object<string, Object|function():Object>}
  */
 const TYPE_CHECK_TARGETS = {
+  // Below are targets containing individual directories which are fully passing
+  // type-checking. Do not remove or disable anything on this list.
+  // Goal: Remove 'QUIET' from all of them.
```
Comment (rsimha) [inline build-system/tasks/check-types.js:63]: Can this be linked via a TODO to an existing issue?

**Review by rsimha — APPROVED** [review]

File: build-system/tasks/check-types.js (hunk 3 above)
Comment (rcebulko) [reply build-system/tasks/check-types.js:63]: We have a new but very in-flux doc at the moment that we plan to make into an I2I after some tweaking. Meanwhile, I'll be actively working in this area in the near future. Would it be alright to link in the issue when it exists?

Comment (rsimha) [reply build-system/tasks/check-types.js:63]: SGTM 👍

File: build-system/tasks/check-types.js (hunk 1 above)
Comment (rcebulko) [reply build-system/tasks/check-types.js:34]: Done

**Pull request merged**